CodeReview#2 of 5 (bullaku) from 10 (Balzan)

TaulantBullaku opened the pull request:
hello, I kindly ask you if you can do the codereview of my code inside ForumLogin.java

Commit:
Fixed web search intent, fixed AutoCompleteTextView covering the button, handled the case intent action is not action search

frankplus committed:
Change text correction code, now including hyphen in word

frankplus committed:
Change text correction code, now including hyphen in word

frankplus committed:
Merge branch 'textAutoCorrection' into wikipedia

frankplus committed:
Merge branch 'search_ingredient_menu' into demo_g3

frankplus committed:
single time list ingredients load using singleton

frankplus committed:
Fix merge conflict

frankplus committed:
Renamed singleton class

frankplus committed:
PhotoTester using singleton for inci loading

frankplus committed:
Merge branch 'galleryActivity' into demo_g3

frankplus committed:
Added search button in ingred details activity.

frankplus committed:
change text correction into a faster and shorter code

frankplus committed:
add a test

frankplus committed:
Merge Brightness_recognition into demo_g3

moved brightness recognition thread into a separate class

frankplus committed:
changed BrightnessRecognitionThread into private

frankplus committed:
Changed BrightnessRecognition thread into an asynctask

frankplus committed:
Changed ingredient details activity into a dialog fragment

Commit:
Implemented search button, added on enter key listener to perform a search, other view improvements

Commit:
Fixed null pointer exception caused by selectedIngredient.getInciName

leopardo-rossi-developer committed:
Forum module and request manager

Creation of a module to handle the forum. Implementation of the class responsible to send network requests

leopardo-rossi-developer committed:
Post model

Implemetation of a class that describes a post object

leopardo-rossi-developer committed:
Viewmodels and fragments

Implementation of the viewmodels and fragment to realize the forum

Commit:
Added autocomplete textview with suggestions

Commit:
Merge remote-tracking branch 'origin/demo_g3' into allergensActivityAutoCompleteImplementation

# Conflicts:
#	app/src/main/AndroidManifest.xml
#	app/src/main/res/values/strings.xml

Commit:
merge with current g3 demo, added on enter-key listener, added drawable icons in allergens activity

thomasporro committed:
Update ResultActivity.java

-Modified Result activity in order to avoid a NullPointerException

PrandiniUniPD committed:
Testing pics section - some bug fixes

stefanoromanello committed:
Show allergen as dropdown onClick

leopardo-rossi-developer committed:
Post layout

Definition of post layout

thomasporro committed:
Added BlurDetection (not working)

-Added blur detection

leopardo-rossi-developer committed:
Get posts

Implementation of the method in show posts viewmodel to get posts from the forum

leopardo-rossi-developer committed:
Posts adapter

Implementation of the adapter to populate the UI with the posts from the forum

frankplus committed:
Merge branch 'restructuring' into demo_g2+g3

BalzPit committed:
Added swipeView feature to AllergensActivity 

To add this new function the main AllergensActivity now only initializes the two Fragments that are used to make the layout. They are the AllergensSearchFragment and UserSAllergensFragment, and they are used to let the user add new allergens and view his own list separately to improve usability. You can either select the tab from the upper bar or swipe to switch between the two fragments.

thomasporro committed:
Update the module

-Added Blur Method

Co-Authored-By: Oscar Garrido <nintenking@users.noreply.github.com>
Co-Authored-By: giofasan <giofasan@users.noreply.github.com>

leopardo-rossi-developer committed:
Comments

Comments addition to a more explicative code

thomasporro committed:
Update ImageProcessing.java

-Added brightness recognition

Co-Authored-By: giofasan <giofasan@users.noreply.github.com>
Co-Authored-By: leopra <leopra@users.noreply.github.com>

leopardo-rossi-developer committed:
Interface update

Updating of the interfaces methods

AlbVal97 committed:
Login feature first commit

AlbVal97 committed:
Login fragment layout completed

AlbVal97 committed:
UI objects initialized

AlbVal97 committed:
Set "Login_VM.java" viewmodel ready to be implemented

AlbVal97 committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForum

# Conflicts:
#	forum/src/main/java/unipd/se18/ocrcamera/forum/viewmodels/Login_VM.java

AlbVal97 committed:
Set "Login_VM.java" viewmodel ready to be implemented (after context-issue)

PrandiniUniPD committed:
AddPost_VM - work started

PrandiniUniPD committed:
AddPost_VM - work started detail

AlbVal97 committed:
Made the fragment ForumLogin interact with its viewmodel Login_VM

leopardo-rossi-developer committed:
Post constructor

Implementation of a new constructor for Post model

frankplus committed:
Avoid hardcoding

thomasporro committed:
Update ImageProcessing.java

-Added AutoAdjustment of brightness

Co-Authored-By: Oscar Garrido <nintenking@users.noreply.github.com>
Co-Authored-By: giofasan <giofasan@users.noreply.github.com>

frankplus committed:
added comments

thomasporro committed:
Update ImageProcessing.java

Co-Authored-By: Oscar Garrido <nintenking@users.noreply.github.com>
Co-Authored-By: giofasan <giofasan@users.noreply.github.com>

frankplus committed:
Added close button to ingredient details fragment

thomasporro committed:
Update module

-Modified interface
-Added comments

Co-Authored-By: Oscar Garrido <nintenking@users.noreply.github.com>
Co-Authored-By: giofasan <giofasan@users.noreply.github.com>

frankplus committed:
using vector asset search icon

frankplus committed:
Merge branch 'search_ingredient_menu' into demo_g3

frankplus committed:
Merge branch 'wikipedia' into demo_g3

frankplus committed:
Merge branch 'textAutoCorrection' into demo_g3

PrandiniUniPD committed:
AddPost_VM - removing method for creating a post, listener for communicating with the View from the ViewModel

PrandiniUniPD committed:
AddPost_VM - details

AlbVal97 committed:
Completed the network request with username and password in Login_VM

leopardo-rossi-developer committed:
Server addition

Add the server url for request in RequestManager

PrandiniUniPD committed:
AddPost_VM - better organization, start of the test's class, testing implementation in build.gradle of the forum module

PrandiniUniPD committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForum

PrandiniUniPD committed:
AddPost_VM - testing is started

PrandiniUniPD committed:
AddPost_VM - testing is started

PrandiniUniPD committed:
AddPost_VM - onJSONPostCreationFailed

PrandiniUniPD committed:
AddPost_VM - onJSONPostCreationFailed(String error)

AlbVal97 committed:
Swapped from email to username and added specifications

AlbVal97 committed:
Implemented interaction between fragment and viewmodel through Observer and LiveData

stefanoromanello committed:
Merge branch 'demo_g3' into galleryActivity

stefanoromanello committed:
Added some Gallery's tests

stefanoromanello committed:
Used allergens found instead of reliability

PrandiniUniPD committed:
AddPost_VM - author parameter

PrandiniUniPD committed:
ForumTest - Some Post tests and some AddPost_VM tests

PrandiniUniPD committed:
AddPost_VM some enhancements for the future code review

PrandiniUniPD committed:
AddPost_VM - JSON failure handled

stefanoromanello committed:
Removed reliability and allergens, implemented AdapterIngredient

AlbVal97 committed:
Completed interaction between fragments using Observers and LiveData

AlbVal97 committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForum

PrandiniUniPD committed:
AddPost_VM - JSON failure handled (better way than before)

stefanoromanello committed:
Comments

PrandiniUniPD committed:
AddPost_VM - detail of a spec

AlbVal97 committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForum

PrandiniUniPD committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForum

GiovanniFurlan committed:
AddPost method first commit

First draft of the method

Commit:
Merge branch 'discussionForum' into discussionForumAddPost

GiovanniFurlan committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForumAddPost

GiovanniFurlan committed:
Merge Conflicts resolved

PrandiniUniPD committed:
AddPost_VM - better name of the internal listener used for notify events to the view

BalzPit committed:
Fixed Allergen Deselection

previously you couldn't deselect allergens once they were selected.
HELP FROM STEFANO ROMANELLO

PrandiniUniPD committed:
AddPost_VM - Checks of parameters (not null and not empty Strings)

PrandiniUniPD committed:
AddPost_VM - Checks of parameters (not null and not empty Strings) - Log

NicoloCervo committed:
allergendb.csv reading fixed

NicoloCervo committed:
Merge remote-tracking branch 'origin/allergensActivityAutoCompleteImplementation' into allergensActivityAutoCompleteImplementation

AlbVal97 committed:
Made the fragment transaction pass the username properly (still using Observers)

leopardo-rossi-developer committed:
Permissions and testability

Addition of internet permission and allow module testability

frankplus committed:
static asynctask and remove unnecessary code

frankplus committed:
Merge branch 'galleryActivity' into demo_g3

GiovanniFurlan committed:
Add Listener in AddPhoto

add listener in addPhoto fragment

Commit:
Added filter to show smart suggestions, added comments

PrandiniUniPD committed:
ForumTest and ForumInstrumentedTest - work started

thomasporro committed:
Modified comments

Nintenking committed:
Merge branch 'ImageProcessing_module' into demo_g1+g2+g3

# Conflicts:
#	.idea/misc.xml
#	.idea/modules.xml
#	app/build.gradle
#	app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java
#	settings.gradle

PrandiniUniPD committed:
ForumTest and ForumInstrumentedTest - other work

PrandiniUniPD committed:
Test - splitting competence

PrandiniUniPD committed:
Test - details about the imports

PrandiniUniPD committed:
AddPost_VMInstrumentedTest - parameters tests

PrandiniUniPD committed:
AddPost_VMInstrumentedTest - parameters tests detail

PrandiniUniPD committed:
AddPost_VMInstrumentedTest - parameters tests detail

leopardo-rossi-developer committed:
Show post listener

Implementation of a listener in substitution of observers

GiovanniFurlan committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForumAddPost

frankplus committed:
Merge branch 'demo_g3' into allergensActivity

GiovanniFurlan committed:
Complete AddPost,debugged fragment_forum_login

Complete AddPost
resolved a bug in fragment_forum_login

GiovanniFurlan committed:
Merge branch 'discussionForumAddPost' into discussionForum

TaulantBullaku committed:
Here the interface for the listener for Login_java

TaulantBullaku committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForum

PrandiniUniPD committed:
Login_VM - indication to support the development of this class (other details of AddPost_VM that remains to commit)

PrandiniUniPD committed:
AddPost_VMTest - Parts relative to the JSON commented

PrandiniUniPD committed:
AddPost_VMTest - JSON work started, misc.xml repaired (break caused by a git stash)

PrandiniUniPD committed:
AddPost_VMTest - Test about parameters and JSON

PrandiniUniPD committed:
AddPost_VM - details for code review

Nintenking committed:
Singleton openCV

PrandiniUniPD committed:
AddPost_VMTest - JSON tests

thomasporro committed:
Solved app crash

-Added class to do PreProcessing
-Solved crash

leopardo-rossi-developer committed:
Database manager class

Implementation of a new class to retrieve data from the forum's database

leopardo-rossi-developer committed:
Parameter correction

leopardo-rossi-developer committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForum

leopardo-rossi-developer committed:
Hardcoding correction

Delete hardcoding from show posts view model

leopardo-rossi-developer committed:
Post layout

Restyling of post layout

frankplus committed:
Log execution time

Commit:
Updated comments

frankplus committed:
Moved allergen manager into singleton and highlight ingr if is allergen

frankplus committed:
highlight yellow if is allergen, red if is an allergen selected by user

frankplus committed:
Fix issue of getSelectedAllergens

Commit:
Added smart filter in SearchResultActivity, code commented

frankplus committed:
Merge remote-tracking branch 'origin/demo_g3' into demo_g3

frankplus committed:
bugfix

frankplus committed:
Merge branch 'demo_g3' into allergensActivity

frankplus committed:
Fix result activity layout

Commit:
Merge remote-tracking branch 'origin/allergensActivityDropdownFilter' into allergensActivityAutoCompleteImplementation

Commit:
merge with allergensActivityDropDownFilter, changed some variables names

Commit:
Fixed not showing allergen selected

Commit:
Fixed removed allergen not removed from view and updated allergen adapter layout

PrandiniUniPD committed:
AddPost and AddPost_VM use DatabaseManager

PrandiniUniPD committed:
google-services.json

frankplus committed:
Corrected ingredients wrongly highlighted

PrandiniUniPD committed:
AddPost_VM - Testing classes adjusted

frankplus committed:
Merge branch 'allergensActivityAutoCompleteImplementation' into allergensActivity

PrandiniUniPD committed:
PostTest - adjusted

frankplus committed:
Merge branch 'allergensActivity' into demo_g3

frankplus committed:
comments

PrandiniUniPD committed:
AddPost_VMIntrumentedTest - adjusted tests

GiovanniFurlan committed:
Minor change in AddPost

Add error string in string.xml in resources
Listener initialization before call to AddPost_VM
If an error happens and the fragment restart it pass the last information wrote trough bundle

GiovanniFurlan committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForum

PrandiniUniPD committed:
AddPost_VMIntrumentedTest - tests with mocking

PrandiniUniPD committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForum

PrandiniUniPD committed:
AddPost_VM - Strings for notifier adjusted

leopardo-rossi-developer committed:
Delete google-services.json

leopardo-rossi-developer committed:
Add files via upload

leopardo-rossi-developer committed:
Automatic post's date conversion

Implementation of a method to convert authomatically a date into the desired format

PrandiniUniPD committed:
AddPost_VM - details

leopardo-rossi-developer committed:
Like addition

Implemetation of the necessary methods to add a like to a specific post

leopardo-rossi-developer committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForum

leopardo-rossi-developer committed:
Add like listener

Implementation of add like listener in the fragment

leopardo-rossi-developer committed:
Post Detail

Implementation of post detail

NicoloCervo committed:
added methods to check lists of ingredients for allergens

stefanoromanello committed:
Load number of allergen in the main page of the gallery

NicoloCervo committed:
added comments

thomasporro committed:
Modified structure of the module

-PreProcessing is now the main class of the module
-Renamed ImageProcessing.java in ExtractTheText.java
-Moved some methods from ExtractTheText to PreProcessing
-Renamed ImageProcessingMethods.java in PreProcessingMethods.java
-Added builder statements to ExtractTheText.java
-Modified method's access of IPBuilder
-IPBuilder now contains static methods

frankplus committed:
Code preparation for code review

Commit:
Update PreProcessingMethods.java

frankplus committed:
comments

thomasporro committed:
Removed null problem from deskew process

leopardo-rossi-developer committed:
Comment addition

Implementation of the code to add a new commet to a specific post

Nintenking committed:
Added Builder for Bitmap's getPixels() and Imgproc.HoughLinesP()

Nintenking committed:
Added import for Bitmap in IPBuilder

thomasporro committed:
Removed hardcoding

frankplus committed:
moved Ucrop into result activity

frankplus committed:
Fixed UI update

frankplus committed:
comments

frankplus committed:
bugfix

frankplus committed:
small change

AlbVal97 committed:
Partial refactoring of ForumLogin and Login_VM to make use of DatabaseManager instead of RequestManager

AlbVal97 committed:
Completed the refactoring of ForumLogin and Login_VM to switch from RequestManager to DatabaseManager

leopardo-rossi-developer committed:
Post detail layout changes

Changes in layout of post detail fragment

leopra committed:
Recognition

leopardo-rossi-developer committed:
Forum module rebuilding

leopra committed:
Recognition

AlbVal97 committed:
Minor changes to ForumLogin and Login_VM, comments and authors fixed

frankplus committed:
reduce false ingredients extracted inside random words

frankplus committed:
separate words correction from text correct (prepare for multithreading)

frankplus committed:
implemented multithreading

frankplus committed:
set minimum number of words per task. Added comments

leopardo-rossi-developer committed:
Module integration

Connection of all the developed fragments so that the module can follow a specif flow

BalzPit commented on forum/src/main/java/unipd/se18/ocrcamera/forum/ForumLogin.java:
```diff
@@ -0,0 +1,174 @@
+    private String userName;
+    private String userPwd;
+
+    private Login_VM viewModel;
```
The global variables are very well commented and easy to understand, except this one (viewModel). I had to look for its class in the other files to know what it was for.

TaulantBullaku replied:
thanks for the clarification, I will comment in more detail. thank you

BalzPit commented on forum/src/main/java/unipd/se18/ocrcamera/forum/ForumLogin.java:
```diff
@@ -0,0 +1,174 @@
+            public void onLoginFailure(String message) {
+
+                if (message != null) {
+                    Toast.makeText(getContext(), message, Toast.LENGTH_LONG).show();
```
Here I can't really understand where the content of the String "message" comes from. Is it something in the "Login_VM" class? I didn't seem to find anything there though.

BalzPit commented on forum/src/main/java/unipd/se18/ocrcamera/forum/ForumLogin.java:
```diff
@@ -0,0 +1,174 @@
+
+    }
+
+    public interface OnFragmentInteractionListener
```
I see that you initialized "mListener" at line 36 but it does not do anything. Does that TODO mean it's a work in progress or it just specifies what the method onFragmentInteraction should do? I also noticed there are three public interfaces with this same name on different classes of the code you sent me. Why is that? Could it be a good idea to write on another file and implement the interface in the classes? Maybe there's something i'm missing though.

leopardo-rossi-developer committed:
Login check

Check login fields before start login process

leopardo-rossi-developer committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForum

BalzPit reviewed (commented):
The code is well written and easy to read. There are a few things I could't quite understand but it's probably because I didn't have the complete view of all the classes and their functions, maybe next time write a small summary of the general function of the class in the descriprion of the pull request. Being a newbie and having used Fragments in my code I also learned how to better and more safely create one.

leopardo-rossi-developer committed:
Login correction

frankplus committed:
Better normalized distance check for word correction

frankplus committed:
Merge branch 'ingredientsExtraction' into demo_g3

frankplus committed:
Merge branch 'demo_g3' into cropOnResultActivity

leopardo-rossi-developer committed:
Login problem solved

Login process now working

GiovanniFurlan committed:
Corrected error in AddPost

When listener pass an error now the fragment is not realoaded
Pass the username when call showpostsfragment after a new post

PrandiniUniPD committed:
Merge remote-tracking branch 'origin/demo_g1+g2+g3' into discussionForum

# Conflicts:
#	.idea/modules.xml
#	app/build.gradle
#	app/src/main/AndroidManifest.xml
#	settings.gradle

PrandiniUniPD committed:
Forum module integrated to the OCRCamera app

PrandiniUniPD committed:
Forum module integrated to the OCRCamera app - Forum manifests adjusted for making only an app

PrandiniUniPD committed:
PostTest updated

PrandiniUniPD committed:
Merging for a CommonDemo - Common style document, MLKitRecognizer with only useful code, start of MlKitRecognizerInstrumentedTest

frankplus committed:
minor changes as suggested in the review

frankplus committed:
Merge branch 'cropOnResultActivity' into demo_g3

thomasporro committed:
Modified the code after the review

-Modified method extractTheText due to a conceptual error
-Modified switch into editBright in PreProcessing.java

thomasporro committed:
Merge branch 'ImageProcessing_module' of https://github.com/PrandiniUniPD/OCRCamera into ImageProcessing_module

frankplus committed:
comments

Nintenking committed:
Added review code for Oscar in PreProcessing.java

thomasporro committed:
Update ResultActivity.java

frankplus committed:
Merge pull request #135 from PrandiniUniPD/ImageProcessing_module

we are merging this, in preparation of the common demo with group 1 and group 2. This merge updates the image processing module.

frankplus committed:
Merge branch 'demo_g3' into demo_g1+g2+g3

# Conflicts:
#	.idea/misc.xml
#	app/build.gradle
#	app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java

frankplus committed:
Added process image in Result Activity

Nintenking committed:
Made the autorotating optional

frankplus committed:
Merge branch 'ImageProcessing_module' into demo_g1+g2+g3

frankplus committed:
changed result activity to comply image processing module change

PrandiniUniPD committed:
Merge remote-tracking branch 'origin/demo_g1+g2+g3' into discussionForum

# Conflicts:
#	.idea/modules.xml
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java
#	app/src/main/res/menu/result_menu.xml
#	app/src/main/res/values/strings.xml

frankplus committed:
bug fix and added image processing into phototester (debug needed)

PrandiniUniPD committed:
Merge remote-tracking branch 'origin/demo_g1+g2+g3' into discussionForum

leopardo-rossi-developer committed:
Integration in main app

leopardo-rossi-developer committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForum

leopardo-rossi-developer committed:
Bug in post's addition

Now when the user add a post, it is visible in the app

AlbVal97 committed:
Fixed code after @Giovanni Furlan code review

leopardo-rossi-developer committed:
Google services json file

leopardo-rossi-developer committed:
Google services JSON file

PrandiniUniPD committed:
ResultActivity - work on device (not on virtual machine)

AlbVal97 committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForum

# Conflicts:
#	.idea/modules.xml

leopardo-rossi-developer committed:
DatabaseManager update

Udate DatabaseManager code with comments and improvements

leopardo-rossi-developer committed:
Merge remote-tracking branch 'origin/discussionForum' into discussionForum

leopardo-rossi-developer committed:
Post and ShowPosts_VM update

Handling date conversion for a Post with a conseguent update of ShowPosts_VM

leopardo-rossi-developer committed:
Code compiles

leopardo-rossi-developer committed:
ShowPost update

The code inside this class has been improved. IN particular harcoding has been avoided

leopardo-rossi-developer committed:
PostDetail_VM update

Code improvements in PostDetail_VM

leopardo-rossi-developer committed:
Improvements

Code improvements

AlbVal97 committed:
Added label to register a new account and its listener, added comments in ForumLogin.java

leopardo-rossi-developer committed:
User's registration to the forum

Implementation of the methods to register a new user to the forum, so that its content can be accessable

AlbVal97 committed:
Added classes and interfaces related to sign up process, partially completed Register_VM view model

AlbVal97 committed:
Completd registration fragment layout, made views ForumRegister and ForumLogin interact

AlbVal97 committed:
Improved Register_VM code readability

AlbVal97 committed:
Improved Register_VM listeners reliability

AlbVal97 committed:
Improved Register_VM readability (part 2)

AlbVal97 committed:
Handled getActivity() issues in ForumRegister

AlbVal97 committed:
Replaced getActivity() and getContext() with requireActivity() and requireContext() respectively and updated ForumRegister

AlbVal97 committed:
ForumRegister and Register_VM classes completed

AlbVal97 committed:
Adjusted ForumRegister response to user interaction, improved code readability
